### PR TITLE
Fix XmlDictionaryReaderQuotas so MaxDepth and MaxStringContentLength are honored

### LINKED ISSUE #4651: XmlDictionaryReaderQuotas.MaxDepth \ MaxStringContentLength don't seem to be honored in dnxcore50 \ coreclr
Minimal repro from a failing test in Mvc:

```C#
using System;
using System.IO;
using System.Runtime.Serialization;
using System.Text;
using System.Xml;

namespace ConsoleApp4
{
    public class Program
    {
        public static void Main(string[] args)
        {
            var input = "<?xml version=\"1.0\" encoding=\"UTF-8\"?><TestLevelTwo><TestOne><SampleInt>10</SampleInt></TestOne></TestLevelTwo>";
            var content = new MemoryStream(Encoding.UTF8.GetBytes(input));
            var quotas = new XmlDictionaryReaderQuotas
            {
                MaxDepth = 1,
            };

            using (var reader = XmlDictionaryReader.CreateTextReader(content, Encoding.UTF8, quotas, onClose: null))
            {
                var serializer = new DataContractSerializer(typeof(TestLevelTwo), new DataContractSerializerSettings());

                // This throws a SerializerException in dnx451 but succeeds just fine in CoreCLR.
                var deserializedObject = (TestLevelTwo)serializer.ReadObject(reader);
                Console.WriteLine(deserializedObject.TestOne.SampleInt);
            }
        }
    }

    [DataContract(Name = "TestLevelOne", Namespace = "")]
    public class TestLevelOne
    {
        [DataMember]
        public int SampleInt { get; set; }
    }

    [DataContract(Name = "TestLevelTwo", Namespace = "")]
    public class TestLevelTwo
    {
        [DataMember]
        public TestLevelOne TestOne { get; set; }
    }
}
```
We're seeing a similar issue with MaxStringContentLength. I'm using build 23509 of `System.Runtime.Serialization.Xml`

### PR DESCRIPTION
**Pull request by khdang**
MaxDepth and MaxStringContentLength settings in XmlDictionaryReaderQuotas are not effective due to missing the right handling in the implementation. This change adds the checks and throws exception when these two limit are exceeded.

Fix #4651 

@shmao @SGuyGe

### CODE REVIEW

**Comment by stephentoub** [comment]
LGTM

**Comment by krwq** [comment]
LGTM

File: src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs [hunk 1]
```diff
@@ -2711,4 +2711,21 @@ public class TypeWithTimeSpanProperty
 public class TypeWithByteProperty
 {
     public byte ByteProperty;
-}
\ No newline at end of file
+}
+
+[DataContract(Name = "TestLevelOne", Namespace = "")]
+public class TestLevelOne
```
Comment (SGuyGe) [inline src/System.Runtime.Serialization.Xml/tests/SerializationTypes.cs]: TestLevelOne is a verb phrase and more suitable for method name than class name, better use a descriptive noun phrase.

**Comment by khdang** [comment]
Test Innerloop Windows_NT Debug Build and Test please

**Comment by khdang** [comment]
Test Innerloop Windows_NT Release Build and Test please

**Pull request merged**